### ISSUE TITLE
Feature Request: shortcut options

### ISSUE THREAD

**Issue by axelson** [dashbitco/nimble_options#78]
As a developer I want to have the ability to treat some options as aliases of others, similar to command line arguments where `-q` could be equivalent to `--quiet`

Specifically in [scenic](https://github.com/boydm/scenic) the goal is to allow developers to specify the translate (as one example) as either `:t` or `:translate`:
https://github.com/boydm/scenic/blob/b2e57b50737443d8fb7a6d7ef2e88ef704cf42c2/lib/scenic/primitive/transform/translate.ex#L21-L29

This is currently implemented using `:rename_to` which is now deprecated in 0.4:
https://github.com/boydm/scenic/blob/b2e57b50737443d8fb7a6d7ef2e88ef704cf42c2/lib/scenic/primitive/transform/transform.ex#L61-L68

It would be nice to be able to do this without having to post-process the output of nimble_options (one specific downside to this is having to customize the output of [NimbleOptions.docs/2](https://hexdocs.pm/nimble_options/NimbleOptions.html#docs/2)).

This is effectively a request to include `:other_names` mentioned in https://github.com/dashbitco/nimble_options/issues/68#issuecomment-953619287 although I think `:aliases` would be a better name for it.

**Comment by whatyouhide** [comment]
@axelson I think aliases (or shortcuts) are much more common in CLI tools arguments rather than options like the ones NimbleOptions is targeting. As you said, doing a pre-pass on the options to rename them before feeding them into nimble_options should be straightforward and, as you mentioned, you can customize the `doc:` option to mention aliases.

You could even do all of this in a programmatic way by preprocessing the schema too:

```elixir
schema_with_aliases =
  Enum.map(schema, fn {key, schema} ->
    case Keyword.fetch(schema, :aliases) do
      {:ok, aliases} ->
        doc = "Aliased to: #{inspect(aliases)}"
        Keyword.update(schema, :doc, " #{doc}", &(&1 <> " #{doc}")
  end)
```

or something along those lines. You get the idea. Thoughts?

**Comment by josevalim** [comment]
Yeah, I think a pre-pass is the way to go too. I understand why it exists on Scenic and agree with it, but we likely don’t want to encourage APIs to have aliases. It is rather the exception than the rule.
